core/src/test/java/org/elasticsearch/indices/analyze/AnalyzeActionIT.java

```diff
@@ -503,4 +503,20 @@ public void testDetailAnalyzeWithMultiValuesWithCustomAnalyzer() throws Exceptio
 
     }
 
+    public void testNonExistTokenizer() {
+        assertAcked(prepareCreate("test").addAlias(new Alias("alias")));
+        ensureGreen();
+
+        try {
+            AnalyzeResponse analyzeResponse = client().admin().indices()
+                .prepareAnalyze("this is a test")
+                .setAnalyzer("not_exist_analyzer")
+                .get();
+            fail("shouldn't get here");
+        } catch (Throwable t) {
+            assertThat(t, instanceOf(IllegalArgumentException.class));
```

Comment:
Just catch IllegalArgumentException I think. Maybe assert something about the message but its not a big deal.